### PR TITLE
Convert CheckDocs in componenttest to command tool

### PR DESCRIPTION
**Pull request by mxiamxia**
**Description:** 
Convert `CheckDocs` from `componenttest` to the standalone command tool and remove the checks from unit test in service module to minimize the exposed API.
- [x] Convert `CheckDocs` to binary command tool in Core repository, add it in `workflow` check and remove related check from unit test
- [ ] Import `CheckDocs` tool in Contrib repository, add it in `workflow` check and remove related check from unit test. (Will be done in separate PR)
- [ ] Delete `doc.go`, `docs.go` and `docs_test.go` files from `componenttest` in Core repository once `cmd/checkdoc` tag is released

**Link to tracking Issue:** 
P1 GA - https://github.com/open-telemetry/opentelemetry-collector/issues/2664

**Testing:** 
`make all`
`make checkdoc`

**Documentation:**
See README

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -22,3 +22,7 @@ updates:
     directory: "/internal/tools"
     schedule:
       interval: "weekly"
+  - package-ecosystem: "gomod"
+    directory: "/cmd/checkdoc"
+    schedule:
+      interval: "weekly"
```
Comment (bogdandrutu) [inline .github/dependabot.yml:28]: Let's for the moment leave it in the main module, since we agreed to not have too many modules.

**Review by bogdandrutu — COMMENTED** [review]
LGTM in general

File: .github/dependabot.yml (hunk 1 above)
Comment (mxiamxia) [reply .github/dependabot.yml:28]: If we need to run CheckDocs binary in `contrib` workflow, I am not sure how to get it installed along there if we leave it in main module.

Comment (bogdandrutu) [reply .github/dependabot.yml:28]: You can install a specific command from a module, I think :)

Comment (bogdandrutu) [reply .github/dependabot.yml:28]: We can do that later

File: Makefile [hunk 2]
```diff
@@ -356,3 +361,8 @@ certs:
 .PHONY: certs-dryrun
 certs-dryrun:
 	@internal/buildscripts/gen-certs.sh -d
+
+# Verify existence of READMEs for components specified as default components in the collector.
+.PHONY: checkdoc
+checkdoc:
+	go run cmd/checkdoc/main.go cmd/checkdoc/docs.go --project-path $(CURDIR) --component-rel-path $(COMP_REL_PATH) --module-name $(MOD_NAME)
```
Comment (bogdandrutu) [inline Makefile:368]: New line

**Review by bogdandrutu — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (mxiamxia) [reply Makefile]: Updated. Thanks!

**Pull request merged**